### PR TITLE
Update and Delete experience

### LINKED ISSUE #10: Delete existing Experience
Using a `DELETE `request for the `/resume/experience` route, delete an existing Experience using its index position as an ID. As part of this, you should write a test in `test_pytest.py` to show it works.

### LINKED ISSUE #21: Edit existing Experience
Using a ```PUT``` request for the ```/resume/experience route```, update an existing Experience using its index position as an ID. As part of this, you should write a test in ```test_pytest.py``` to show it works.

### PR DESCRIPTION
**Pull request by demarillacizere**
- DELETE request for deleting a specific experience.
- PUT request for deleting a specific experience.
- Respective test functions for both requests.

Closes #10 
Closes #21

### CODE REVIEW

**Comment by demarillacizere** [comment]
@dtemir This PR is ready for review.

File: app.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def hello_world():
     return jsonify({"message": "Hello, World!"})
 
 @app.route('/resume/experience', methods=['GET', 'POST'])
-@app.route('/resume/experience/<index>', methods=['GET', 'POST'])
+@app.route('/resume/experience/<index>', methods=['GET', 'POST','DELETE','pUT'])
```
Comment (dtemir) [inline app.py:61]: Let's fix the typo

File: app.py [hunk 2]
```diff
@@ -104,6 +104,47 @@ def experience(index = None):
             "message": "New experience successfully created",
             "id": len(data['experience']) -1
             }), 201
+    if request.method == 'DELETE':
+        if index and str(index).isnumeric():
+                expId = int(index)
```
Comment (dtemir) [inline app.py:109]: can you pleas follow the python convention of [snake_case](https://en.wikipedia.org/wiki/Snake_case)? i.e. `exp_id` like on line 121

File: app.py [hunk 3]
```diff
@@ -104,6 +104,47 @@ def experience(index = None):
...
+
+            # Request validation
+            required_fields = ['title', 'company', 'start_date', 'description', 'logo']
+            if body.get('start_date') and not validate_date_string(body.get('start_date')):
+                return jsonify({"error": "Invalid start date. The format should be `June 2023`"}), 400
+            if(body.get('end_date') and not validate_date_string(body.get('end_date'))):
+                return jsonify({"error": "Invalid end date. Format should be like `June 2023`"}), 400
+            if not validate_request(body, required_fields):
+                return jsonify({"error": "Invalid request. Required attributes are missing"}), 400
+            # Request validation End
```
Comment (dtemir) [inline app.py:135]: Looks good, but do you think we could abstract this part away like Suelen did in #38?

File: test_pytest.py [hunk 4]
```diff
@@ -90,6 +90,35 @@ def test_add_experience():
...
+            "title": "Senior Software Developer",
+            "company": "A very Cool Company",
+            "start_date": "October 2022",
+            "end_date": "October 2024",
+            "description": "Writing Python Code",
+            "logo": "example-logo.png",
+        },
+    )
+                                     
+     # Assert the response status code is 200 (OK)
```
Comment (dtemir) [inline test_pytest.py:108]: 201

**Review by dtemir — CHANGES_REQUESTED** [review]

File: app.py (hunk 2 above)
Comment (demarillacizere) [reply app.py]: Done

**Comment by demarillacizere** [comment]
@dtemir Please review again

**Comment by dtemir** [comment]
Hey @demarillacizere! Thanks for fixing the suggestions.

Do you know what happened to pytests by chance? The last PR that I merged #41 seem to show them as green but they're failing now. Can I ask you to look into this? If this is failing because of someone else's PR, let's raise an issue and ask them to resolve it.

**Comment by demarillacizere** [comment]
> Hey @demarillacizere! Thanks for fixing the suggestions.
> 
> Do you know what happened to pytests by chance? The last PR that I merged #41 seem to show them as green but they're failing now. Can I ask you to look into this? If this is failing because of someone else's PR, let's raise an issue and ask them to resolve it.

It was an issue with  the 'phonenumbers' package but now the tests are passing.

**Comment by dtemir** [comment]
Sweet! Great work.

**Comment by demarillacizere** [comment]
closes #20

**Pull request merged**